sdk/e2e/src/main/java/com/azure/endtoend/identity/MultiTenantTest.java

```diff
@@ -0,0 +1,164 @@
+
+import java.io.IOException;
+import java.util.Locale;
+
+/**
+ * Runs Identity tests across multiple tests.
+ */
+class MultiTenantTest {
+    private static final String AZURE_MULTI_TENANT_TEST_MODE = "AZURE_MULTI_TENANT_TEST_MODE";
+    private static final String AZURE_USER_NAME = "AZURE_USER_NAME";
+    private static final String AZURE_USER_PASSWORD = "AZURE_USER_PASSWORD";
+    private static final String AZURE_CLIENT_ID = "AZURE_CLIENT_ID";
+    private static final String AZURE_CLIENT_SECRET = "AZURE_CLIENT_SECRET";
+    private static final String AZURE_TENANT_ID = "AZURE_TENANT_ID";
+    private static final Configuration CONFIGURATION = Configuration.getGlobalConfiguration().clone();
+    private final ClientLogger logger = new ClientLogger(MultiTenantTest.class);
+
+    /**
+     * Runs the multi tenant identity tests
+     * @throws IllegalStateException if AZURE_MULTI_TENANT_TEST_MODE is not set to "user" or "sp"
+     */
+    void run() throws IllegalStateException {
+        if (CoreUtils.isNullOrEmpty(CONFIGURATION.get(AZURE_MULTI_TENANT_TEST_MODE))) {
+            throw logger.logExceptionAsError(new IllegalStateException("Test mode is not set. Set environment "
+                                                + "variable AZURE_MULTI_TENANT_TEST_MODE to user or sp"));
+        }
+
+        String mode = CONFIGURATION.get(AZURE_MULTI_TENANT_TEST_MODE).toLowerCase(Locale.ENGLISH);
+        switch (mode) {
+            case "user":
+                testUserPasswordCanAccessGraph();
+                break;
+            case "sp":
+                testServicePrincipalCanAccessGraph();
+                break;
+            default:
+                throw logger.logExceptionAsError(
+                    new IllegalStateException("Invalid Test mode is configured AZURE_MULTI_TENANT_TEST_MODE. "
+                                                    + "Possible values are user or sp."));
+        }
+    }
+
+    private void testUserPasswordCanAccessGraph() {
+        assertConfigPresence(AZURE_CLIENT_ID,
+            "testUserPasswordCanAccessGraph - AZURE_CLIENT_ID not configured in the environment.");
+        assertConfigPresence(AZURE_TENANT_ID,
+            "testUserPasswordCanAccessGraph - AZURE_TENANT_ID not configured in the environment.");
+        assertConfigPresence(AZURE_USER_NAME,
+            "testUserPasswordCanAccessGraph - AZURE_USER_NAME not configured in the environment.");
+        assertConfigPresence(AZURE_USER_PASSWORD,
+            "testUserPasswordCanAccessGraph - AZURE_USER_PASSWORD not configured in the environment.");
+        String clientId = CONFIGURATION.get(AZURE_CLIENT_ID);
+        String tenantId = CONFIGURATION.get(AZURE_TENANT_ID);
+        String username = CONFIGURATION.get(AZURE_USER_NAME);
+        String password = CONFIGURATION.get(AZURE_USER_PASSWORD);
+
+        UsernamePasswordCredential credential = new UsernamePasswordCredentialBuilder()
+            .clientId(clientId)
```

Comment:
so, your instructions cover to disable 2FA for the username ?

Reply:
This has to be done in a separate tenant - cannot be done from the Microsoft corporate tenant. By default 2FA is turned off in any new tenant.